### DISCUSSION TITLE
How do I convert jaxpr to hlo

### DISCUSSION THREAD

**Discussion by songh11** [jax-ml/jax#15647] [Q&A] [answered]
Hi, I just got in touch with jax, but I have a problem.
I built a model and can generate jaxpr using jax.make_jaxpr, but I encountered a error when using jax.jit().lower().

Does jax have an interface to convert jaxpr directly to hlo?

thanks

**Comment by jakevdp** [comment]
The main way to generate hlo is to use `jax.jit(f).lower(*args)`. If you're able to include a small reproduction of the problematic code, or at the very least a traceback showing the error, we may be able to help figure out what's going wrong

**Reply by songh11** [reply]
Many thanks, I used a framework developed based on jax, so I can't show my code.
The error is displayed as shown in the figure:
(![share](https://user-images.githubusercontent.com/75419275/232949434-19437634-b54d-497f-bccd-a2d3b9d93e38.jpg))

**Reply by jakevdp** [reply, marked as answer]
An `ArgInfo` object cannot be converted into an array, so this error looks like the correct output. With the lack of context, it's hard to give any advice better than "don't do that".

**Reply by songh11** [reply]
I think it's the type of parameters inside my model, I'll find out why
Thank you very much for your help
